moped-editor/src/queries/components.js

```diff
@@ -130,6 +133,48 @@ export const UPDATE_COMPONENT_ATTRIBUTES = gql`
   }
 `;
 
+// This mutation updates component subcomponents in same way as UPDATE_COMPONENT_ATTRIBUTES.
+// It also updates all component feature_signals to is_deleted = true and then inserts the new signal.
+export const UPDATE_SIGNAL_COMPONENT = gql`
```

Comment:
this autocomplete feature really does throw in so many complications - thank you for sorting through them!